vars/cloudFoundryDeploy.groovy

```diff
@@ -196,11 +196,19 @@ private String deleteOptionIfRequired(Map config) {
     }
 }
 
-private String stopOldAppIfRequired(Map config) {
+private void stopOldAppIfRequired(Map config) {
+    String oldAppName = "${config.cloudFoundry.appName}-old"
+    String cfStopOutputFileName = 'cfStopOutput.txt'
+
     if (config.keepOldInstance && config.deployType == 'blue-green') {
-        return "cf stop ${config.cloudFoundry.appName}-old"
-    } else {
-        return ''
+        int cfStopReturncode = sh (returnStatus: true, script: "cf stop $oldAppName  > $cfStopOutputFileName")
```

Comment:
In case it is really required to rely on log file parsing (see other comment below) I would like to suggest to redirect also stderr and not only stdout in order to ensure to have the full log output in the file:

`cf stop $oldAppName &> $cfStopOutputFileName`

Normally log output should be served via `stderr` anyway in order to keep `stdout` free for "payload" which can be piped to other commands, but this is a different story.

Reply:
Agree. Will add the `&`